pkg/secrets/keyvault/client.go

```diff
@@ -160,31 +160,28 @@ func (k *KeyvaultSecretClient) Create(ctx context.Context, key types.NamespacedN
 			}
 		}
 		// If flatten has not been declared, convert the map into a json string for persistance
-	} else {
-		jsonData, err := json.Marshal(data)
-		if err != nil {
-			return err
-		}
-		stringSecret := string(jsonData)
-		contentType := "json"
-
-		// Initialize secret parameters
-		secretParams := keyvaults.SecretSetParameters{
-			Value:            &stringSecret,
-			SecretAttributes: &secretAttributes,
-			ContentType:      &contentType,
-		}
-
-		if _, err := k.KeyVaultClient.GetSecret(ctx, vaultBaseURL, secretBaseName, secretVersion); err == nil {
-			return fmt.Errorf("secret already exists %v", err)
-		}
-
-		_, err = k.KeyVaultClient.SetSecret(ctx, vaultBaseURL, secretBaseName, secretParams)
-
+	}
+	jsonData, err := json.Marshal(data)
+	if err != nil {
 		return err
 	}
+	stringSecret := string(jsonData)
+	contentType := "json"
+
+	// Initialize secret parameters
+	secretParams := keyvaults.SecretSetParameters{
+		Value:            &stringSecret,
+		SecretAttributes: &secretAttributes,
+		ContentType:      &contentType,
+	}
 
-	return nil
+	if _, err := k.KeyVaultClient.GetSecret(ctx, vaultBaseURL, secretBaseName, secretVersion); err == nil {
+		return fmt.Errorf("secret already exists %v", err)
```

Comment:
Suggest using `%w` to wrap the error
```suggestion
		return fmt.Errorf("secret already exists: %w", err)
```